lib/core-classes/AssetService.ts

```diff
@@ -109,11 +98,24 @@ export class AssetService {
   // @todo remove when we have the date extractor in the core
   private iso8601 (value: string, name: string): Date {
     const parsed: any = new Date(value);
-
     if (isNaN(parsed)) {
       throw new BadRequestError(`"${name}" is not a valid ISO8601 date`);
     }
-
     return parsed;
   }
+
+
+  async getDeviceByAsset(assetId :string) {
+    const filter = {
+      'query': {
+        'term': {
+          'assetId': {
+            'value': assetId
+          }
+        }
+      }
+    } ;
+    return await this.sdk.document.search('device-manager', 'devices', filter);
```

Comment:
Useless await.

Also, you should use the Koncorde syntax when it's possible because it will be portable between storage backend in the futur

Reply:
this search is not useful in the current version and is deleted.